src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPool2.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Concurrent;
+
+namespace Microsoft.AspNet.Server.Kestrel.Infrastructure
+{
+    /// <summary>
+    /// Used to allocate and distribute re-usable blocks of memory.
+    /// </summary>
+    public class MemoryPool2 : IDisposable
+    {
+        /// <summary>
+        /// The gap between blocks' starting address. 4096 is chosen because most operating systems are 4k pages in size and alignment.
+        /// </summary>
+        private const int blockStride = 4096;
```

Comment:
The other instances of `private const` in Kestrel are still prefixed with "_" (e.g. `MemoryPoolTextWriter._textLength`)